### PR TITLE
LPS-168310 Redirect patterns: disable save btn when there is an error

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
Added validation for Destination URL field. Whenever there is an error, the save button is disabled. 

Screenshot 2022-11-14 at 13 55 25

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by ambrinchaudhary** [comment]
@AliciaGarciaGarcia I've added here what Brian suggested in https://github.com/brianchandotcom/liferay-portal/pull/126182#issuecomment-1313524550 

@boton please review. I finally go ahead with having a separated Set for invalid urls, that has also to be "sync" when a row is deleted and it contained an error.

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

File: modules/apps/redirect/redirect-web/src/main/java/com/liferay/redirect/web/internal/display/context/RedirectPatternConfigurationDisplayContext.java [hunk 1]
```diff
@@ -58,6 +58,8 @@ public String getRedirectPatternConfigurationURL() {
 
 	public Map<String, Object> getRedirectPatterns() {
 		return HashMapBuilder.<String, Object>put(
+			"actionUrl", getRedirectPatternConfigurationURL()
```
Comment (AliciaGarciaGarcia) [inline modules/apps/redirect/redirect-web/src/main/java/com/liferay/redirect/web/internal/display/context/RedirectPatternConfigurationDisplayContext.java:61]: I could not found another place than here that use : `getRedirectPatternConfigurationURL`
Suggestion: 
Maybe change it to be private is better?

File: modules/apps/redirect/redirect-web/src/main/resources/META-INF/resources/js/RedirectPatterns.js [hunk 2]
```diff
@@ -22,14 +22,24 @@ import uuidv4 from 'uuid/v4';
...
 	index,
 	pattern = '',
 	portletNamespace,
 }) => {
+	const [destinationUrl, setDestinationUrl] = useState(initialDestinationUrl);
+	const [urlError, setUrlError] = useState(false);
+
+	const urlAllowRelative = (url) => {
+		return REGEX_URL_ALLOW_RELATIVE && REGEX_URL_ALLOW_RELATIVE.test(url);
+	};
```
Comment (boton) [inline modules/apps/redirect/redirect-web/src/main/resources/META-INF/resources/js/RedirectPatterns.js:41]: Can we move this fn outside? We don't need to construct it on each render

Comment (boton) [reply modules/apps/redirect/redirect-web/src/main/resources/META-INF/resources/js/RedirectPatterns.js:41]: We can make it shorter too

```js
const urlAllowRelative = (url) => REGEX_URL_ALLOW_RELATIVE?.test(url);
```

Or even, if we don't want to check our own constant.

```js
const urlAllowRelative = (url) => REGEX_URL_ALLOW_RELATIVE.test(url);
```

**Review by boton — CHANGES_REQUESTED** [review]
We have a problem with the new `invalidPatternsSet` using the index isn't working.


https://user-images.githubusercontent.com/67901/201912919-8df793de-ba6d-48f6-ad64-0c7eea8638af.mp4

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

**Comment by ambrinchaudhary** [comment]
ci:reevaluate:1297401_5472

**Comment by adolfopa** [comment]
ci:reevaluate:1297401_5472

**Comment by adolfopa** [comment]
ci:test:relevant

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Review by boton — APPROVED** [review]
Ambrin and I add some SF updates here [41156da](https://github.com/liferay-lima/liferay-portal/pull/3712/commits/41156da6aaf621b4e310623ebe784e7782f65446).

LGTM but not tested yet

**Comment by ambrinchaudhary** [comment]
ci:forward

**Pull request closed without merging**